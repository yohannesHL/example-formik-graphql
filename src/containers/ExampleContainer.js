import React, { useState } from "react";
import ExampleForm from "../components/ExampleForm";
import { useMutation, useQuery } from "@apollo/client";
import { GET_FORM } from "../graphql/queries";
import { SET_FORM } from "../graphql/mutations";

export default function ExampleContainer() {
  const [mutate] = useMutation(SET_FORM);
  const { data, refetch } = useQuery(GET_FORM);
  const [state, setState] = useState(
    data ?? {
      userName: "",
      email: "",
    }
  );

  async function handleSubmit(values) {
    await mutate({
      variables: { userName: values.userName, email: values.email },
    });
    const query = await refetch();
    setState(query.data);
  }

  return (
    <div>
      <h2 role="heading">Form</h2>
      <ExampleForm initialValues={state} onSubmit={handleSubmit} />
    </div>
  );
}
