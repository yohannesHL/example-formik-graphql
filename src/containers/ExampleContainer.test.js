import React from "react";
import * as apollo from "@apollo/client";
import ExampleContainer from "./ExampleContainer";
import { render, act } from "@testing-library/react";
import { MockedProvider } from "@apollo/client/testing";

it("runs the mocked query", async () => {
  await act(async () => {
    jest.spyOn(apollo, "useQuery").mockReturnValue({
      data: {
        userName: "TestUser",
        email: "testuser@example.com",
      },
      refetch: jest.fn(),
    });
    jest.spyOn(apollo, "useMutation");

    const { container } = await render(
      <MockedProvider>
        <ExampleContainer />
      </MockedProvider>
    );
    await new Promise((resolve) => setTimeout(resolve, 0)); // wait for response

    expect(container).toMatchSnapshot();
  });
});
