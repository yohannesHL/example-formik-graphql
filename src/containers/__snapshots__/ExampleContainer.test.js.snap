// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`runs the mocked query 1`] = `
<div>
  <div>
    <h2
      role="heading"
    >
      Form
    </h2>
    <form
      action="#"
      class="ExampleForm"
    >
      <label
        for="userName"
      >
        Username: 
      </label>
      <input
        aria-label="userName-input"
        id="userName"
        name="userName"
        style="border: 1px solid;"
        type="text"
        value="TestUser"
      />
      <label
        for="email"
      >
        Email: 
      </label>
      <input
        aria-label="email-input"
        id="email"
        name="email"
        type="text"
        value="testuser@example.com"
      />
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
